packages/test-interface-criteria/src/criteria/criteriaUtils.js

```diff
@@ -36,25 +35,30 @@ function assertNodeConstraintType(expectedType, requirement, criteriaName) {
  * @template {TestDataNodeConstraint} TNodeConstraint
  * @template {keyof TNodeConstraint} TFieldName
  * @param {TFieldName} fieldName
- * @param {TNodeConstraint} reqA
- * @param {TNodeConstraint} reqB
+ * @param {TNodeConstraint} baseConstraint
+ * @param {TNodeConstraint} extensionConstraint
  * @param {(reqAField: TNodeConstraint[TFieldName], reqBField: TNodeConstraint[TFieldName]) => TNodeConstraint[TFieldName] | null} getValueIfBothExist
  * @returns {{} | Pick<TNodeConstraint, TFieldName>} This format makes it easy to merge this data into an object literal.
  *   The result will look like e.g. `{ mininclusive: 3 }`.
  *   It can also be an empty object to cater for instances in which neither of the requirements have the field
  *   and therefore this field should not be added to the merged requirement.
  */
-function mergeTestDataNodeConstraintField(fieldName, reqA, reqB, getValueIfBothExist) {
-  if (reqA[fieldName] == null && reqB[fieldName] == null) {
+function mergeTestDataNodeConstraintField(
+  fieldName,
+  baseConstraint,
+  extensionConstraint,
+  getValueIfBothExist
+) {
+  if (baseConstraint[fieldName] == null && extensionConstraint[fieldName] == null) {
     return {};
   }
-  if (reqA[fieldName] != null) {
-    return { [fieldName]: reqA[fieldName] };
+  if (extensionConstraint[fieldName] != null && baseConstraint[fieldName] == null) {
+    return { [fieldName]: extensionConstraint[fieldName] };
   }
-  if (reqB[fieldName] != null) {
-    return { [fieldName]: reqB[fieldName] };
+  if (baseConstraint[fieldName] != null && extensionConstraint[fieldName] == null) {
+    return { [fieldName]: baseConstraint[fieldName] };
```

Comment:
This seems to me to have been an error. Why have the `getValueIfBothExist(..)` invokation down below if we'll never reach that because we would return at the first condition.